diff --git a/sys-kernel/linux-firmware/linux-firmware-9999.ebuild b/sys-kernel/linux-firmware/linux-firmware-9999.ebuild
index 4e7daf6..49c0a11 100644
--- a/sys-kernel/linux-firmware/linux-firmware-9999.ebuild
+++ b/sys-kernel/linux-firmware/linux-firmware-9999.ebuild
@@ -146,7 +146,7 @@ src_install() {
 	use_fw adsp_skl && doins_subdir intel/dsp_fw_*
 	use_fw ath9k_htc && doins htc_*.fw
 	use_fw ath10k && doins_subdir ath10k/QCA6174/hw3.0/*
-	use_fw bcm4354-bt && doins_subdir brcm/BCM4354_*.hcd
+	use_fw bcm4354-bt && doins_subdir brcm/BCM4354_*.hcd && doins_subdir brcm/bcm*
 	use_fw cros-pd && doins_subdir cros-pd/*
 	use_fw fw_sst && doins_subdir intel/fw_sst*
 	use_fw fw_sst2 && doins_subdir intel/IntcSST2.bin
@@ -189,7 +189,7 @@ src_install() {
 	for x in "${IUSE_BRCMWIFI[@]}"; do
 		use_fw ${x} || continue
 		case ${x} in
-		brcmfmac-all)      doins_subdir brcm/brcmfmac* ;;
+		brcmfmac-all)      doins_subdir brcm/brcmfmac* && doins_subdir brcm/bcm*;;
 		brcmfmac4354-sdio) doins_subdir brcm/brcmfmac4354-sdio.* ;;
 		brcmfmac4356-pcie) doins_subdir brcm/brcmfmac4356-pcie.* ;;
 		brcmfmac4371-pcie) doins_subdir brcm/brcmfmac4371-pcie.* ;;
