diff --git a/chromeos-base/crosh/crosh-9999.ebuild b/chromeos-base/crosh/crosh-9999.ebuild
index 895f774..5c5cc80 100644
--- a/chromeos-base/crosh/crosh-9999.ebuild
+++ b/chromeos-base/crosh/crosh-9999.ebuild
@@ -44,7 +44,7 @@ src_compile() {
 }
 
 src_install() {
-	dobin crosh crosh-{dev,usb}
+	dobin crosh crosh-{dev,usb,custom}
 	dobin network_diag
 	insinto /usr/share/misc
 	doins "${WORKDIR}"/inputrc.crosh
