diff --git a/login_manager/init/scripts/ui-pre-start b/login_manager/init/scripts/ui-pre-start
index 780b469..06e2feb 100755
--- a/login_manager/init/scripts/ui-pre-start
+++ b/login_manager/init/scripts/ui-pre-start
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+/opt/myscript/inspect_hw.sh
+
 # Copyright 2016 The Chromium OS Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
