diff --git a/drivers/gpu/drm/i915/intel_bios.c b/drivers/gpu/drm/i915/intel_bios.c
index 673d351..6c6b4f2 100644
--- a/drivers/gpu/drm/i915/intel_bios.c
+++ b/drivers/gpu/drm/i915/intel_bios.c
@@ -302,8 +302,10 @@ parse_lfp_backlight(struct drm_i915_private *dev_priv, struct bdb_header *bdb)
 	const struct bdb_lfp_backlight_data_entry *entry;
 
 	backlight_data = find_section(bdb, BDB_LVDS_BACKLIGHT);
-	if (!backlight_data && IS_VALLEYVIEW(dev_priv->dev)) {
-		fake_vlv_backlight(dev_priv);
+	if (!backlight_data){
+		if (IS_VALLEYVIEW(dev_priv->dev)) {
+			fake_vlv_backlight(dev_priv);
+		}
 		return;
 	}
 
