diff --git a/virtual/target-chromium-os-dev/target-chromium-os-dev-1.ebuild b/virtual/target-chromium-os-dev/target-chromium-os-dev-1.ebuild
index 5448b3c..7e4ffa1 100644
--- a/virtual/target-chromium-os-dev/target-chromium-os-dev-1.ebuild
+++ b/virtual/target-chromium-os-dev/target-chromium-os-dev-1.ebuild
@@ -11,7 +11,7 @@ SLOT="0"
 KEYWORDS="*"
 # Note: Do not utilize USE=internal here.  Update virtual/target-chrome-os-dev.
 IUSE="cras pam opengl +power_management +profile
-	+shill tpm usb video_cards_intel"
+	+shill tpm usb video_cards_intel X"
 
 # The dependencies here are meant to capture "all the packages
 # developers want to use for development, test, or debug".  This
@@ -30,7 +30,7 @@ IUSE="cras pam opengl +power_management +profile
 
 ################################################################################
 #
-# CROS_* : Dependencies for CrOS devices (coreutils, etc.)
+# CROS_* : Dependencies for CrOS devices (coreutils, X etc)
 #
 ################################################################################
 CROS_X86_RDEPEND="
@@ -42,9 +42,31 @@ CROS_X86_RDEPEND="
 	video_cards_intel? ( x11-apps/intel-gpu-tools )
 "
 
+CROS_X_RDEPEND="
+	opengl? ( x11-apps/mesa-progs )
+	x11-apps/mtplot
+	x11-apps/xdpyinfo
+	opengl? ( x11-apps/xdriinfo )
+	x11-apps/xev
+	x11-apps/xhost
+	x11-apps/xinput
+	x11-apps/xinput_calibrator
+	x11-apps/xlsatoms
+	x11-apps/xlsclients
+	x11-apps/xmodmap
+	x11-apps/xprop
+	x11-apps/xset
+	!x11-apps/xtrace
+	x11-apps/xwd
+	x11-apps/xwininfo
+	x11-misc/xdotool
+	x11-misc/xtrace
+"
+
 RDEPEND="
 	x86? ( ${CROS_X86_RDEPEND} )
 	amd64? ( ${CROS_X86_RDEPEND} )
+	X? ( ${CROS_X_RDEPEND} )
 "
 
 RDEPEND="${RDEPEND}
diff --git a/virtual/target-chromium-os-test/target-chromium-os-test-1.ebuild b/virtual/target-chromium-os-test/target-chromium-os-test-1.ebuild
index e81fe36..08de20c 100644
--- a/virtual/target-chromium-os-test/target-chromium-os-test-1.ebuild
+++ b/virtual/target-chromium-os-test/target-chromium-os-test-1.ebuild
@@ -12,7 +12,7 @@ SLOT="0"
 KEYWORDS="*"
 # Note: Do not utilize USE=internal here.  Update virtual/target-chrome-os-test.
 IUSE="-chromeless_tests chromeless_tty cros_ec cros_embedded opengl opengles ozone p2p
-	peerd +shill +tpm"
+	peerd +shill +tpm X"
 
 # Packages required to support autotest images.  Dependencies here
 # are for packages that must be present on a local device and that
@@ -77,7 +77,7 @@ CROS_COMMON_RDEPEND+="
 
 ################################################################################
 #
-# CROS_* : Dependencies for "regular" CrOS devices (coreutils, etc.)
+# CROS_* : Dependencies for "regular" CrOS devices (coreutils, X etc)
 #
 ################################################################################
 CROS_X86_RDEPEND="
@@ -85,9 +85,19 @@ CROS_X86_RDEPEND="
 	x11-misc/read-edid
 "
 
+CROS_X_RDEPEND="
+	x11-apps/setxkbmap
+	x11-apps/xinput
+	x11-apps/xset
+	x11-misc/x11vnc
+	x11-misc/xdotool
+	x11-terms/rxvt-unicode
+"
+
 CROS_RDEPEND="
 	x86? ( ${CROS_X86_RDEPEND} )
 	amd64? ( ${CROS_X86_RDEPEND} )
+	X? ( ${CROS_X_RDEPEND} )
 "
 
 CROS_RDEPEND="${CROS_RDEPEND}
@@ -165,6 +175,18 @@ CROS_RDEPEND="${CROS_RDEPEND}
 	"
 
 ################################################################################
+# CROS_E_* : Dependencies for embedded CrOS devices (busybox, no X etc)
+#
+################################################################################
+
+#CROS_E_RDEPEND="${CROS_E_RDEPEND}
+#"
+
+# Build time dependencies
+#CROS_E_DEPEND="${CROS_E_RDEPEND}
+#"
+
+################################################################################
 # Assemble the final RDEPEND and DEPEND variables for portage
 ################################################################################
 RDEPEND="${CROS_COMMON_RDEPEND}
diff --git a/virtual/target-chromium-os/target-chromium-os-1.ebuild b/virtual/target-chromium-os/target-chromium-os-1.ebuild
index d5ed737..8b608a5 100644
--- a/virtual/target-chromium-os/target-chromium-os-1.ebuild
+++ b/virtual/target-chromium-os/target-chromium-os-1.ebuild
@@ -56,6 +56,7 @@ IUSE="
 	watchdog
 	wifi_bootstrapping
 	wimax
+	X
 "
 
 REQUIRED_USE="cellular? ( shill )"
@@ -110,8 +111,8 @@ REQUIRED_USE="cellular? ( shill )"
 # Variable Naming Convention:
 # ---------------------------
 # CROS_COMMON_* : Dependencies common to all CrOS flavors
-# CROS_E_* : Dependencies for embedded CrOS devices (busybox, etc.)
-# CROS_* : Dependencies for "regular" CrOS devices (coreutils, etc.)
+# CROS_E_* : Dependencies for embedded CrOS devices (busybox, no X etc)
+# CROS_* : Dependencies for "regular" CrOS devices (coreutils, X etc)
 ################################################################################
 
 ################################################################################
@@ -238,10 +239,17 @@ CROS_ARM_RDEPEND="
 	chromeos-base/u-boot-scripts
 "
 
+CROS_X_RDEPEND="
+	chromeos-base/xorg-conf
+	x11-apps/xrandr
+	x11-base/xorg-server
+"
+
 CROS_RDEPEND="
 	x86? ( ${CROS_X86_RDEPEND} )
 	amd64? ( ${CROS_X86_RDEPEND} )
 	arm? ( ${CROS_ARM_RDEPEND} )
+	X? ( ${CROS_X_RDEPEND} )
 "
 
 CROS_RDEPEND="${CROS_RDEPEND}
@@ -297,7 +305,7 @@ CROS_DEPEND="${CROS_RDEPEND}
 "
 
 ################################################################################
-# CROS_E_* : Dependencies for embedded CrOS devices (busybox, etc.)
+# CROS_E_* : Dependencies for embedded CrOS devices (busybox, no X etc)
 #
 ################################################################################
 
