diff --git a/x11-drivers/xf86-input-vmmouse/xf86-input-vmmouse-13.1.0-r1.ebuild b/x11-drivers/xf86-input-vmmouse/xf86-input-vmmouse-13.1.0-r1.ebuild
index 397451c8..38ff0150 100644
--- a/x11-drivers/xf86-input-vmmouse/xf86-input-vmmouse-13.1.0-r1.ebuild
+++ b/x11-drivers/xf86-input-vmmouse/xf86-input-vmmouse-13.1.0-r1.ebuild
@@ -3,7 +3,7 @@
 
 EAPI=5
 
-inherit udev xorg-2
+inherit xorg-2
 
 DESCRIPTION="VMWare mouse input driver"
 IUSE=""
@@ -18,7 +18,6 @@ pkg_setup() {
 		--with-hal-bin-dir=/punt
 		--with-hal-callouts-dir=/punt
 		--with-hal-fdi-dir=/punt
-		--with-udev-rules-dir=$(get_udevdir)/rules.d
 	)
 
 	xorg-2_pkg_setup
