diff --git a/content/browser/compositor/gpu_process_transport_factory.cc b/content/browser/compositor/gpu_process_transport_factory.cc
index 874debf..ef14437 100644
--- a/content/browser/compositor/gpu_process_transport_factory.cc
+++ b/content/browser/compositor/gpu_process_transport_factory.cc
@@ -197,10 +197,11 @@ static bool ShouldCreateGpuOutputSurface(ui::Compositor* compositor) {
   }
 #endif
 
-#if defined(OS_CHROMEOS)
-  // Software fallback does not happen on Chrome OS.
-  return true;
-#endif
+// enable Software fallback on Chrome OS for EEE PC 900.
+//#if defined(OS_CHROMEOS)
+//  // Software fallback does not happen on Chrome OS.
+//  return true;
+//#endif
 
 #if defined(OS_WIN)
   if (::GetProp(compositor->widget(), kForceSoftwareCompositor) &&
@@ -254,10 +254,10 @@ void GpuProcessTransportFactory::EstablishedGpuChannel(
   DCHECK(data);
 
   if (num_attempts > kNumRetriesBeforeSoftwareFallback) {
-#if defined(OS_CHROMEOS)
-    LOG(FATAL) << "Unable to create a UI graphics context, and cannot use "
-               << "software compositing on ChromeOS.";
-#endif
+//#if defined(OS_CHROMEOS)
+//    LOG(FATAL) << "Unable to create a UI graphics context, and cannot use "
+//               << "software compositing on ChromeOS.";
+//#endif
     create_gpu_output_surface = false;
   }
diff --git a/content/browser/gpu/gpu_process_host.cc b/content/browser/gpu/gpu_process_host.cc
index 9c72997..80de8f9 100644
--- a/content/browser/gpu/gpu_process_host.cc
+++ b/content/browser/gpu/gpu_process_host.cc
@@ -1041,12 +1041,13 @@ void GpuProcessHost::RecordProcessCrash() {
       if ((gpu_recent_crash_count_ >= kGpuMaxCrashCount &&
            !disable_crash_limit) ||
           !initialized_) {
-#if !defined(OS_CHROMEOS)
+	// enable falldown to software rendering with CHROMEOS
+//#if !defined(OS_CHROMEOS)
         // The GPU process is too unstable to use. Disable it for current
         // session.
         hardware_gpu_enabled_ = false;
         GpuDataManagerImpl::GetInstance()->DisableHardwareAcceleration();
-#endif
+//#endif
       }
     }
   }
