diff --git a/ui/display/chromeos/display_configurator.cc b/ui/display/chromeos/display_configurator.cc
index e6327b4..4c3fb68 100644
--- a/ui/display/chromeos/display_configurator.cc
+++ b/ui/display/chromeos/display_configurator.cc
@@ -970,17 +970,6 @@ void DisplayConfigurator::SuspendDisplays(
   // Stop |configure_timer_| because we will force probe and configure all the
   // displays at resume time anyway.
   configure_timer_.Stop();
-
-  // Turn off the displays for suspend. This way, if we wake up for lucid sleep,
-  // the displays will not turn on (all displays should be off for lucid sleep
-  // unless explicitly requested by lucid sleep code). Use
-  // SetDisplayPowerInternal so requested_power_state_ is maintained.
-  SetDisplayPowerInternal(chromeos::DISPLAY_POWER_ALL_OFF,
-                          kSetDisplayPowerNoFlags, callback);
-
-  // We need to make sure that the monitor configuration we just did actually
-  // completes before we return.
-  native_display_delegate_->SyncWithServer();
 }
 
 void DisplayConfigurator::ResumeDisplays() {
