diff --git a/BUILD.gn b/BUILD.gn
index 797f473..4605401 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -261,8 +261,8 @@ static_library("libvpx") {
       ":libvpx_yasm",
       ":libvpx_intrinsics_mmx",
       ":libvpx_intrinsics_sse2",
-      ":libvpx_intrinsics_ssse3",
-      ":libvpx_intrinsics_sse4_1",
+#      ":libvpx_intrinsics_ssse3",
+#      ":libvpx_intrinsics_sse4_1",
     ]
   }
   if (cpu_arch_full == "arm-neon-cpu-detect") {
diff --git a/libvpx_srcs.gni b/libvpx_srcs.gni
index 6c1a1b0..6465a04 100644
--- a/libvpx_srcs.gni
+++ b/libvpx_srcs.gni
@@ -320,15 +320,10 @@ libvpx_srcs_x86_assembly = [
   "//third_party/libvpx/source/libvpx/vp8/common/x86/recon_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_sse3.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_sse4.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/subpixel_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/subpixel_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/subpixel_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_impl_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_impl_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_impl_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/dct_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/dct_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/encodeopt.asm",
@@ -338,19 +333,13 @@ libvpx_srcs_x86_assembly = [
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/subtract_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_copy_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_intrapred_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_intrapred_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_loopfilter_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_8t_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_8t_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_bilinear_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_bilinear_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_dct_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_error_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad4d_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_sse3.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_sse4.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_subpel_variance.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_subtract_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_temporal_filter_apply_sse2.asm",
@@ -378,13 +367,8 @@ libvpx_srcs_x86_sse2 = [
 libvpx_srcs_x86_sse3 = [
 ]
 libvpx_srcs_x86_ssse3 = [
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_ssse3.c",
-  "//third_party/libvpx/source/libvpx/vp8/encoder/x86/quantize_ssse3.c",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_idct_intrin_ssse3.c",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_8t_intrin_ssse3.c",
 ]
 libvpx_srcs_x86_sse4_1 = [
-  "//third_party/libvpx/source/libvpx/vp8/encoder/x86/quantize_sse4.c",
 ]
 libvpx_srcs_x86_avx = [
 ]
@@ -708,15 +692,10 @@ libvpx_srcs_x86_64_assembly = [
   "//third_party/libvpx/source/libvpx/vp8/common/x86/recon_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_sse3.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_sse4.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/sad_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/subpixel_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/subpixel_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/subpixel_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_impl_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_impl_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_impl_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/dct_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/dct_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/encodeopt.asm",
@@ -726,23 +705,14 @@ libvpx_srcs_x86_64_assembly = [
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/subtract_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp8/encoder/x86/subtract_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_copy_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_idct_ssse3_x86_64.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_intrapred_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_intrapred_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_loopfilter_mmx.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_8t_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_8t_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_bilinear_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_bilinear_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_dct_mmx.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_dct_ssse3_x86_64.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_error_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_quantize_ssse3_x86_64.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad4d_sse2.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_sse2.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_sse3.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_sse4.asm",
-  "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_sad_ssse3.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_ssim_opt_x86_64.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_subpel_variance.asm",
   "//third_party/libvpx/source/libvpx/vp9/encoder/x86/vp9_subtract_sse2.asm",
@@ -771,13 +741,8 @@ libvpx_srcs_x86_64_sse2 = [
 libvpx_srcs_x86_64_sse3 = [
 ]
 libvpx_srcs_x86_64_ssse3 = [
-  "//third_party/libvpx/source/libvpx/vp8/common/x86/variance_ssse3.c",
-  "//third_party/libvpx/source/libvpx/vp8/encoder/x86/quantize_ssse3.c",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_idct_intrin_ssse3.c",
-  "//third_party/libvpx/source/libvpx/vp9/common/x86/vp9_subpixel_8t_intrin_ssse3.c",
 ]
 libvpx_srcs_x86_64_sse4_1 = [
-  "//third_party/libvpx/source/libvpx/vp8/encoder/x86/quantize_sse4.c",
 ]
 libvpx_srcs_x86_64_avx = [
 ]
diff --git a/libvpx_srcs_x86_64.gypi b/libvpx_srcs_x86_64.gypi
index f29c6c8..bf298b3 100644
--- a/libvpx_srcs_x86_64.gypi
+++ b/libvpx_srcs_x86_64.gypi
@@ -83,15 +83,15 @@
     '<(libvpx_source)/vp8/common/x86/recon_sse2.asm',
     '<(libvpx_source)/vp8/common/x86/sad_mmx.asm',
     '<(libvpx_source)/vp8/common/x86/sad_sse2.asm',
-    '<(libvpx_source)/vp8/common/x86/sad_sse3.asm',
-    '<(libvpx_source)/vp8/common/x86/sad_sse4.asm',
-    '<(libvpx_source)/vp8/common/x86/sad_ssse3.asm',
+#    '<(libvpx_source)/vp8/common/x86/sad_sse3.asm',
+#    '<(libvpx_source)/vp8/common/x86/sad_sse4.asm',
+#    '<(libvpx_source)/vp8/common/x86/sad_ssse3.asm',
     '<(libvpx_source)/vp8/common/x86/subpixel_mmx.asm',
     '<(libvpx_source)/vp8/common/x86/subpixel_sse2.asm',
-    '<(libvpx_source)/vp8/common/x86/subpixel_ssse3.asm',
+#    '<(libvpx_source)/vp8/common/x86/subpixel_ssse3.asm',
     '<(libvpx_source)/vp8/common/x86/variance_impl_mmx.asm',
     '<(libvpx_source)/vp8/common/x86/variance_impl_sse2.asm',
-    '<(libvpx_source)/vp8/common/x86/variance_impl_ssse3.asm',
+#    '<(libvpx_source)/vp8/common/x86/variance_impl_ssse3.asm',
     '<(libvpx_source)/vp8/common/x86/vp8_asm_stubs.c',
     '<(libvpx_source)/vp8/decoder/dboolhuff.c',
     '<(libvpx_source)/vp8/decoder/dboolhuff.h',
@@ -218,14 +218,14 @@
     '<(libvpx_source)/vp9/common/x86/vp9_asm_stubs.c',
     '<(libvpx_source)/vp9/common/x86/vp9_copy_sse2.asm',
     '<(libvpx_source)/vp9/common/x86/vp9_idct_intrin_sse2.h',
-    '<(libvpx_source)/vp9/common/x86/vp9_idct_ssse3_x86_64.asm',
+#    '<(libvpx_source)/vp9/common/x86/vp9_idct_ssse3_x86_64.asm',
     '<(libvpx_source)/vp9/common/x86/vp9_intrapred_sse2.asm',
-    '<(libvpx_source)/vp9/common/x86/vp9_intrapred_ssse3.asm',
+#    '<(libvpx_source)/vp9/common/x86/vp9_intrapred_ssse3.asm',
     '<(libvpx_source)/vp9/common/x86/vp9_loopfilter_mmx.asm',
     '<(libvpx_source)/vp9/common/x86/vp9_subpixel_8t_sse2.asm',
-    '<(libvpx_source)/vp9/common/x86/vp9_subpixel_8t_ssse3.asm',
+#    '<(libvpx_source)/vp9/common/x86/vp9_subpixel_8t_ssse3.asm',
     '<(libvpx_source)/vp9/common/x86/vp9_subpixel_bilinear_sse2.asm',
-    '<(libvpx_source)/vp9/common/x86/vp9_subpixel_bilinear_ssse3.asm',
+#    '<(libvpx_source)/vp9/common/x86/vp9_subpixel_bilinear_ssse3.asm',
     '<(libvpx_source)/vp9/decoder/vp9_decodeframe.c',
     '<(libvpx_source)/vp9/decoder/vp9_decodeframe.h',
     '<(libvpx_source)/vp9/decoder/vp9_decodemv.c',
@@ -310,14 +310,14 @@
     '<(libvpx_source)/vp9/encoder/vp9_writer.c',
     '<(libvpx_source)/vp9/encoder/vp9_writer.h',
     '<(libvpx_source)/vp9/encoder/x86/vp9_dct_mmx.asm',
-    '<(libvpx_source)/vp9/encoder/x86/vp9_dct_ssse3_x86_64.asm',
+#    '<(libvpx_source)/vp9/encoder/x86/vp9_dct_ssse3_x86_64.asm',
     '<(libvpx_source)/vp9/encoder/x86/vp9_error_sse2.asm',
-    '<(libvpx_source)/vp9/encoder/x86/vp9_quantize_ssse3_x86_64.asm',
+#    '<(libvpx_source)/vp9/encoder/x86/vp9_quantize_ssse3_x86_64.asm',
     '<(libvpx_source)/vp9/encoder/x86/vp9_sad4d_sse2.asm',
     '<(libvpx_source)/vp9/encoder/x86/vp9_sad_sse2.asm',
-    '<(libvpx_source)/vp9/encoder/x86/vp9_sad_sse3.asm',
-    '<(libvpx_source)/vp9/encoder/x86/vp9_sad_sse4.asm',
-    '<(libvpx_source)/vp9/encoder/x86/vp9_sad_ssse3.asm',
+#    '<(libvpx_source)/vp9/encoder/x86/vp9_sad_sse3.asm',
+#    '<(libvpx_source)/vp9/encoder/x86/vp9_sad_sse4.asm',
+#    '<(libvpx_source)/vp9/encoder/x86/vp9_sad_ssse3.asm',
     '<(libvpx_source)/vp9/encoder/x86/vp9_ssim_opt_x86_64.asm',
     '<(libvpx_source)/vp9/encoder/x86/vp9_subpel_variance.asm',
     '<(libvpx_source)/vp9/encoder/x86/vp9_subtract_sse2.asm',
diff --git a/libvpx_srcs_x86_64_intrinsics.gypi b/libvpx_srcs_x86_64_intrinsics.gypi
index bb1b203..79dc896 100644
--- a/libvpx_srcs_x86_64_intrinsics.gypi
+++ b/libvpx_srcs_x86_64_intrinsics.gypi
@@ -51,13 +51,13 @@
         '<(libvpx_source)',
       ],
       'sources': [
-        '<(libvpx_source)/vp8/common/x86/variance_ssse3.c',
-        '<(libvpx_source)/vp8/encoder/x86/quantize_ssse3.c',
-        '<(libvpx_source)/vp9/common/x86/vp9_idct_intrin_ssse3.c',
-        '<(libvpx_source)/vp9/common/x86/vp9_subpixel_8t_intrin_ssse3.c',
+#        '<(libvpx_source)/vp8/common/x86/variance_ssse3.c',
+#        '<(libvpx_source)/vp8/encoder/x86/quantize_ssse3.c',
+#        '<(libvpx_source)/vp9/common/x86/vp9_idct_intrin_ssse3.c',
+#        '<(libvpx_source)/vp9/common/x86/vp9_subpixel_8t_intrin_ssse3.c',
       ],
-      'cflags': [ '-mssse3', ],
-      'xcode_settings': { 'OTHER_CFLAGS': [ '-mssse3' ] },
+#      'cflags': [ '-mssse3', ],
+#      'xcode_settings': { 'OTHER_CFLAGS': [ '-mssse3' ] },
       'conditions': [
         ['OS=="win" and clang==1', {
           # cl.exe's /arch flag doesn't have a setting for SSSE3/4, and cl.exe
@@ -76,10 +76,10 @@
         '<(libvpx_source)',
       ],
       'sources': [
-        '<(libvpx_source)/vp8/encoder/x86/quantize_sse4.c',
+#        '<(libvpx_source)/vp8/encoder/x86/quantize_sse4.c',
       ],
-      'cflags': [ '-msse4.1', ],
-      'xcode_settings': { 'OTHER_CFLAGS': [ '-msse4.1' ] },
+#      'cflags': [ '-msse4.1', ],
+#      'xcode_settings': { 'OTHER_CFLAGS': [ '-msse4.1' ] },
       'conditions': [
         ['OS=="win" and clang==1', {
           # cl.exe's /arch flag doesn't have a setting for SSSE3/4, and cl.exe
diff --git a/libvpx_srcs_x86_intrinsics.gypi b/libvpx_srcs_x86_intrinsics.gypi
index bb1b203..79dc896 100644
--- a/libvpx_srcs_x86_intrinsics.gypi
+++ b/libvpx_srcs_x86_intrinsics.gypi
@@ -51,13 +51,13 @@
         '<(libvpx_source)',
       ],
       'sources': [
-        '<(libvpx_source)/vp8/common/x86/variance_ssse3.c',
-        '<(libvpx_source)/vp8/encoder/x86/quantize_ssse3.c',
-        '<(libvpx_source)/vp9/common/x86/vp9_idct_intrin_ssse3.c',
-        '<(libvpx_source)/vp9/common/x86/vp9_subpixel_8t_intrin_ssse3.c',
+#        '<(libvpx_source)/vp8/common/x86/variance_ssse3.c',
+#        '<(libvpx_source)/vp8/encoder/x86/quantize_ssse3.c',
+#        '<(libvpx_source)/vp9/common/x86/vp9_idct_intrin_ssse3.c',
+#        '<(libvpx_source)/vp9/common/x86/vp9_subpixel_8t_intrin_ssse3.c',
       ],
-      'cflags': [ '-mssse3', ],
-      'xcode_settings': { 'OTHER_CFLAGS': [ '-mssse3' ] },
+#      'cflags': [ '-mssse3', ],
+#      'xcode_settings': { 'OTHER_CFLAGS': [ '-mssse3' ] },
       'conditions': [
         ['OS=="win" and clang==1', {
           # cl.exe's /arch flag doesn't have a setting for SSSE3/4, and cl.exe
@@ -76,10 +76,10 @@
         '<(libvpx_source)',
       ],
       'sources': [
-        '<(libvpx_source)/vp8/encoder/x86/quantize_sse4.c',
+#        '<(libvpx_source)/vp8/encoder/x86/quantize_sse4.c',
       ],
-      'cflags': [ '-msse4.1', ],
-      'xcode_settings': { 'OTHER_CFLAGS': [ '-msse4.1' ] },
+#      'cflags': [ '-msse4.1', ],
+#      'xcode_settings': { 'OTHER_CFLAGS': [ '-msse4.1' ] },
       'conditions': [
         ['OS=="win" and clang==1', {
           # cl.exe's /arch flag doesn't have a setting for SSSE3/4, and cl.exe
diff --git a/source/config/linux/ia32/vp8_rtcd.h b/source/config/linux/ia32/vp8_rtcd.h
index 3fe9d07..afce123 100644
--- a/source/config/linux/ia32/vp8_rtcd.h
+++ b/source/config/linux/ia32/vp8_rtcd.h
@@ -543,7 +543,7 @@ static void setup_rtcd_internal(void)
     if (flags & HAS_SSE3) vp8_diamond_search_sad = vp8_diamond_search_sadx4;
     vp8_fast_quantize_b = vp8_fast_quantize_b_c;
     if (flags & HAS_SSE2) vp8_fast_quantize_b = vp8_fast_quantize_b_sse2;
-    if (flags & HAS_SSSE3) vp8_fast_quantize_b = vp8_fast_quantize_b_ssse3;
+//    if (flags & HAS_SSSE3) vp8_fast_quantize_b = vp8_fast_quantize_b_ssse3;
     vp8_filter_by_weight16x16 = vp8_filter_by_weight16x16_c;
     if (flags & HAS_SSE2) vp8_filter_by_weight16x16 = vp8_filter_by_weight16x16_sse2;
     vp8_filter_by_weight8x8 = vp8_filter_by_weight8x8_c;
@@ -603,7 +603,7 @@ static void setup_rtcd_internal(void)
     if (flags & HAS_SSE3) vp8_refining_search_sad = vp8_refining_search_sadx4;
     vp8_regular_quantize_b = vp8_regular_quantize_b_c;
     if (flags & HAS_SSE2) vp8_regular_quantize_b = vp8_regular_quantize_b_sse2;
-    if (flags & HAS_SSE4_1) vp8_regular_quantize_b = vp8_regular_quantize_b_sse4_1;
+//    if (flags & HAS_SSE4_1) vp8_regular_quantize_b = vp8_regular_quantize_b_sse4_1;
     vp8_sad16x16 = vp8_sad16x16_c;
     if (flags & HAS_MMX) vp8_sad16x16 = vp8_sad16x16_mmx;
     if (flags & HAS_SSE2) vp8_sad16x16 = vp8_sad16x16_wmt;
@@ -668,29 +668,29 @@ static void setup_rtcd_internal(void)
     vp8_sixtap_predict16x16 = vp8_sixtap_predict16x16_c;
     if (flags & HAS_MMX) vp8_sixtap_predict16x16 = vp8_sixtap_predict16x16_mmx;
     if (flags & HAS_SSE2) vp8_sixtap_predict16x16 = vp8_sixtap_predict16x16_sse2;
-    if (flags & HAS_SSSE3) vp8_sixtap_predict16x16 = vp8_sixtap_predict16x16_ssse3;
+//    if (flags & HAS_SSSE3) vp8_sixtap_predict16x16 = vp8_sixtap_predict16x16_ssse3;
     vp8_sixtap_predict4x4 = vp8_sixtap_predict4x4_c;
     if (flags & HAS_MMX) vp8_sixtap_predict4x4 = vp8_sixtap_predict4x4_mmx;
-    if (flags & HAS_SSSE3) vp8_sixtap_predict4x4 = vp8_sixtap_predict4x4_ssse3;
+//    if (flags & HAS_SSSE3) vp8_sixtap_predict4x4 = vp8_sixtap_predict4x4_ssse3;
     vp8_sixtap_predict8x4 = vp8_sixtap_predict8x4_c;
     if (flags & HAS_MMX) vp8_sixtap_predict8x4 = vp8_sixtap_predict8x4_mmx;
     if (flags & HAS_SSE2) vp8_sixtap_predict8x4 = vp8_sixtap_predict8x4_sse2;
-    if (flags & HAS_SSSE3) vp8_sixtap_predict8x4 = vp8_sixtap_predict8x4_ssse3;
+//    if (flags & HAS_SSSE3) vp8_sixtap_predict8x4 = vp8_sixtap_predict8x4_ssse3;
     vp8_sixtap_predict8x8 = vp8_sixtap_predict8x8_c;
     if (flags & HAS_MMX) vp8_sixtap_predict8x8 = vp8_sixtap_predict8x8_mmx;
     if (flags & HAS_SSE2) vp8_sixtap_predict8x8 = vp8_sixtap_predict8x8_sse2;
-    if (flags & HAS_SSSE3) vp8_sixtap_predict8x8 = vp8_sixtap_predict8x8_ssse3;
+//    if (flags & HAS_SSSE3) vp8_sixtap_predict8x8 = vp8_sixtap_predict8x8_ssse3;
     vp8_sub_pixel_mse16x16 = vp8_sub_pixel_mse16x16_c;
     if (flags & HAS_MMX) vp8_sub_pixel_mse16x16 = vp8_sub_pixel_mse16x16_mmx;
     if (flags & HAS_SSE2) vp8_sub_pixel_mse16x16 = vp8_sub_pixel_mse16x16_wmt;
     vp8_sub_pixel_variance16x16 = vp8_sub_pixel_variance16x16_c;
     if (flags & HAS_MMX) vp8_sub_pixel_variance16x16 = vp8_sub_pixel_variance16x16_mmx;
     if (flags & HAS_SSE2) vp8_sub_pixel_variance16x16 = vp8_sub_pixel_variance16x16_wmt;
-    if (flags & HAS_SSSE3) vp8_sub_pixel_variance16x16 = vp8_sub_pixel_variance16x16_ssse3;
+//    if (flags & HAS_SSSE3) vp8_sub_pixel_variance16x16 = vp8_sub_pixel_variance16x16_ssse3;
     vp8_sub_pixel_variance16x8 = vp8_sub_pixel_variance16x8_c;
     if (flags & HAS_MMX) vp8_sub_pixel_variance16x8 = vp8_sub_pixel_variance16x8_mmx;
     if (flags & HAS_SSE2) vp8_sub_pixel_variance16x8 = vp8_sub_pixel_variance16x8_wmt;
-    if (flags & HAS_SSSE3) vp8_sub_pixel_variance16x8 = vp8_sub_pixel_variance16x8_ssse3;
+//    if (flags & HAS_SSSE3) vp8_sub_pixel_variance16x8 = vp8_sub_pixel_variance16x8_ssse3;
     vp8_sub_pixel_variance4x4 = vp8_sub_pixel_variance4x4_c;
     if (flags & HAS_MMX) vp8_sub_pixel_variance4x4 = vp8_sub_pixel_variance4x4_mmx;
     if (flags & HAS_SSE2) vp8_sub_pixel_variance4x4 = vp8_sub_pixel_variance4x4_wmt;
diff --git a/source/config/linux/ia32/vp9_rtcd.h b/source/config/linux/ia32/vp9_rtcd.h
index 660b652..b125d0f 100644
--- a/source/config/linux/ia32/vp9_rtcd.h
+++ b/source/config/linux/ia32/vp9_rtcd.h
@@ -886,22 +886,22 @@ static void setup_rtcd_internal(void)
     if (flags & HAS_SSE2) vp9_block_error = vp9_block_error_sse2;
     vp9_convolve8 = vp9_convolve8_c;
     if (flags & HAS_SSE2) vp9_convolve8 = vp9_convolve8_sse2;
-    if (flags & HAS_SSSE3) vp9_convolve8 = vp9_convolve8_ssse3;
+//    if (flags & HAS_SSSE3) vp9_convolve8 = vp9_convolve8_ssse3;
     vp9_convolve8_avg = vp9_convolve8_avg_c;
     if (flags & HAS_SSE2) vp9_convolve8_avg = vp9_convolve8_avg_sse2;
-    if (flags & HAS_SSSE3) vp9_convolve8_avg = vp9_convolve8_avg_ssse3;
+//    if (flags & HAS_SSSE3) vp9_convolve8_avg = vp9_convolve8_avg_ssse3;
     vp9_convolve8_avg_horiz = vp9_convolve8_avg_horiz_c;
     if (flags & HAS_SSE2) vp9_convolve8_avg_horiz = vp9_convolve8_avg_horiz_sse2;
-    if (flags & HAS_SSSE3) vp9_convolve8_avg_horiz = vp9_convolve8_avg_horiz_ssse3;
+//    if (flags & HAS_SSSE3) vp9_convolve8_avg_horiz = vp9_convolve8_avg_horiz_ssse3;
     vp9_convolve8_avg_vert = vp9_convolve8_avg_vert_c;
     if (flags & HAS_SSE2) vp9_convolve8_avg_vert = vp9_convolve8_avg_vert_sse2;
-    if (flags & HAS_SSSE3) vp9_convolve8_avg_vert = vp9_convolve8_avg_vert_ssse3;
+//    if (flags & HAS_SSSE3) vp9_convolve8_avg_vert = vp9_convolve8_avg_vert_ssse3;
     vp9_convolve8_horiz = vp9_convolve8_horiz_c;
     if (flags & HAS_SSE2) vp9_convolve8_horiz = vp9_convolve8_horiz_sse2;
-    if (flags & HAS_SSSE3) vp9_convolve8_horiz = vp9_convolve8_horiz_ssse3;
+//    if (flags & HAS_SSSE3) vp9_convolve8_horiz = vp9_convolve8_horiz_ssse3;
     vp9_convolve8_vert = vp9_convolve8_vert_c;
     if (flags & HAS_SSE2) vp9_convolve8_vert = vp9_convolve8_vert_sse2;
-    if (flags & HAS_SSSE3) vp9_convolve8_vert = vp9_convolve8_vert_ssse3;
+//    if (flags & HAS_SSSE3) vp9_convolve8_vert = vp9_convolve8_vert_ssse3;
     vp9_convolve_avg = vp9_convolve_avg_c;
     if (flags & HAS_SSE2) vp9_convolve_avg = vp9_convolve_avg_sse2;
     vp9_convolve_copy = vp9_convolve_copy_c;
@@ -989,12 +989,12 @@ static void setup_rtcd_internal(void)
     if (flags & HAS_SSSE3) vp9_h_predictor_8x8 = vp9_h_predictor_8x8_ssse3;
     vp9_idct16x16_10_add = vp9_idct16x16_10_add_c;
     if (flags & HAS_SSE2) vp9_idct16x16_10_add = vp9_idct16x16_10_add_sse2;
-    if (flags & HAS_SSSE3) vp9_idct16x16_10_add = vp9_idct16x16_10_add_ssse3;
+//    if (flags & HAS_SSSE3) vp9_idct16x16_10_add = vp9_idct16x16_10_add_ssse3;
     vp9_idct16x16_1_add = vp9_idct16x16_1_add_c;
     if (flags & HAS_SSE2) vp9_idct16x16_1_add = vp9_idct16x16_1_add_sse2;
     vp9_idct16x16_256_add = vp9_idct16x16_256_add_c;
     if (flags & HAS_SSE2) vp9_idct16x16_256_add = vp9_idct16x16_256_add_sse2;
-    if (flags & HAS_SSSE3) vp9_idct16x16_256_add = vp9_idct16x16_256_add_ssse3;
+//    if (flags & HAS_SSSE3) vp9_idct16x16_256_add = vp9_idct16x16_256_add_ssse3;
     vp9_idct32x32_1024_add = vp9_idct32x32_1024_add_c;
     if (flags & HAS_SSE2) vp9_idct32x32_1024_add = vp9_idct32x32_1024_add_sse2;
     vp9_idct32x32_1_add = vp9_idct32x32_1_add_c;
diff --git a/source/config/linux/ia32/vpx_config.asm b/source/config/linux/ia32/vpx_config.asm
index b47dd1f..7e0730c 100644
--- a/source/config/linux/ia32/vpx_config.asm
+++ b/source/config/linux/ia32/vpx_config.asm
@@ -14,9 +14,9 @@
 %define HAVE_MMX 1
 %define HAVE_SSE 1
 %define HAVE_SSE2 1
-%define HAVE_SSE3 1
-%define HAVE_SSSE3 1
-%define HAVE_SSE4_1 1
+%define HAVE_SSE3 0
+%define HAVE_SSSE3 0
+%define HAVE_SSE4_1 0
 %define HAVE_AVX 1
 %define HAVE_AVX2 0
 %define HAVE_ALTIVEC 0
diff --git a/source/config/linux/ia32/vpx_config.h b/source/config/linux/ia32/vpx_config.h
index 58bda73..e271e1a 100644
--- a/source/config/linux/ia32/vpx_config.h
+++ b/source/config/linux/ia32/vpx_config.h
@@ -26,9 +26,9 @@
 #define HAVE_MMX 1
 #define HAVE_SSE 1
 #define HAVE_SSE2 1
-#define HAVE_SSE3 1
-#define HAVE_SSSE3 1
-#define HAVE_SSE4_1 1
+#define HAVE_SSE3 0
+#define HAVE_SSSE3 0
+#define HAVE_SSE4_1 0
 #define HAVE_AVX 1
 #define HAVE_AVX2 0
 #define HAVE_ALTIVEC 0
