diff --git a/bin/cros_make_image_bootable b/bin/cros_make_image_bootable
index 98f8b69..c3c1cdd 100755
--- a/bin/cros_make_image_bootable
+++ b/bin/cros_make_image_bootable
@@ -289,9 +289,10 @@ make_image_bootable() {
     info "DEBUG: use recovery keyblock"
   fi
 
-  if [ ${FLAGS_force_developer_mode} -eq ${FLAGS_TRUE} ]; then
+  # customize: set cros_debug as defauilt for touble shooting
+  #if [ ${FLAGS_force_developer_mode} -eq ${FLAGS_TRUE} ]; then
     FLAGS_boot_args="${FLAGS_boot_args} cros_debug"
-  fi
+  #fi
 
   # Builds the kernel partition image.
   local rootfs_fs_size=$(get_filesystem_size ${FLAGS_image_type} 3)
