--- a/overlay-x86-xorg/profiles/base/make.defaults	2017-06-24 11:20:55.369633050 +0900
+++ b/overlay-x86-xorg/profiles/base/make.defaults	2017-06-24 11:19:49.762275913 +0900
@@ -11,7 +11,7 @@
 
 USE="${USE} containers device_jail"
 
-USE="${USE} legacy_keyboard legacy_power_button kernel-4_4"
+USE="${USE} legacy_keyboard legacy_power_button kernel-4_4 fbconsole vtconsole"
 
 # Enable all optional features on generic overlays.
 USE="${USE} buffet peerd wifi_bootstrapping"
